hathor/cli/run_node.py

```diff
@@ -232,13 +246,71 @@ def register_signal_handlers(self) -> None:
         if sigusr1 is not None:
             # USR1 is available in this OS.
             signal.signal(sigusr1, self.signal_usr1_handler)
+        sigusr2 = getattr(signal, 'SIGUSR2', None)
+        if sigusr2 is not None:
+            # USR1 is available in this OS.
+            signal.signal(sigusr2, self.signal_usr2_handler)
 
     def signal_usr1_handler(self, sig: int, frame: Any) -> None:
         """Called when USR1 signal is received."""
         self.log.warn('USR1 received. Killing all connections...')
         if self.manager and self.manager.connections:
             self.manager.connections.disconnect_all_peers(force=True)
 
+    def signal_usr2_handler(self, sig: int, frame: Any) -> None:
+        """Called when USR1 signal is received."""
+        from hathor.sysctl.exception import (
+            SysctlEntryNotFound,
+            SysctlException,
+            SysctlReadOnlyEntry,
+            SysctlRunnerException,
+            SysctlWriteOnlyEntry,
+        )
+
+        self.log.warn('USR2 received.')
+
+        if self._args.data:
+            basedir = self._args.data
+            tempdir = None
+        else:
+            basedir = tempfile.mkdtemp()
+            tempdir = basedir
+
+        filename = os.path.join(basedir, f'SIGUSR2-{os.getpid()}.pipe')
+        if os.path.exists(filename):
+            self.log.warn('[USR2] Pipe already exists. Delete it and try again.', pipe=filename)
+            return
+
+        runner = self.get_sysctl_runner()
+
+        with temp_fifo(filename, tempdir):
+            self.log.warn('[USR2] Waiting for commands...', pipe=filename)
+            fp = open(filename, 'r')
+            lines = fp.readlines()
```

Comment:
I considered to keep the pipe open until the user decides to delete it. But it would require non-blocking reading so I skipped it for now. What do you think?

Reply:
We decided to keep it this way for now.